recipes/security_realm_file.rb

```diff
@@ -0,0 +1,45 @@
+#
+# Cookbook Name:: cdap
+# Recipe:: security_realm_file
+#
+# Copyright © 2013-2016 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Manage Authentication realmfile
+if node['cdap']['cdap_site']['security.authentication.basic.realmfile']
+  realmfile = node['cdap']['cdap_site']['security.authentication.basic.realmfile']
+  realmdir = ::File.dirname(realmfile)
+
+  # Ensure parent directory exists
+  unless Dir.exists? realmfile
```

Comment:
This will be evaluated at compile time, but something may create it at converge time before this runs.